### ISSUE TITLE
the training loss is too big and never changed

### ISSUE THREAD

**Issue by youmi-zym** [mrharicot/monodepth#124]
Hello mrharicot:
    Thanks for sharing your code ,it's really helpful!
    I have changed the net which designed by myself. The net is pretrained with the ground truth of KITTI,  and the loss function is the L1. the final error rate is (30%,<5pix).
   And then I change the loss function to  yours . Therefore, only the net changed. 
   I write the code in Pytorch, with the Pytorch asked the loss should be scalar. So I sum up the loss( **total_loss**.sum() ) to a scalar. However, the loss become too big ( batch size is 2, the loss is 1572846, learning rate is 1e-4) , and it changed within [-2,2] even after 500 iterations. Is it ok?

**Comment by mrharicot** [comment]
Hi,
I am not sure I am following what the problem is.
Have you reimplemented our model in pytorch?
If so I'm afraid I won't be able to help you

**Comment by youmi-zym** [comment]
Sorry, my English is poor.  
I have reimplemented the calculation of loss in pytorch and use it .
 And just now, I run the **monodepth_simple** with a KITTI image and get the disp numpy array. With the numpy array, I find the ouput of your **image_loss** is big too,  786432.5625.

`

        self.image_loss=self.image_loss_right+self.image_loss_left
        self.image_loss=self.image_loss.sum()

> `self.total_loss=self.image_loss+config['disp_gradient_loss_weight']*self.disp_loss+self.lr_loss*config['lr_loss_weight']`

`

so ,my question is: 

- the **total_loss** is so big, is that normal

- and with so big loss, will the **'training rate=1e-4'** be ok? or how can I training it, my loss doesn't fall after 500 iterations

Thanks a lot!

**Comment by mrharicot** [comment]
Hi,
The image loss should be the average of the per pixel losses, so at worse it should be 1.
Use:
`self.image_loss=self.image_loss.mean()`
If you are using adam, a learning rate of 1e-4 is usually fine.

I hope this helps!

**Comment by youmi-zym** [comment]
Wow, thanks a lot! my mistake! It works now.

**Comment by mrharicot** [comment]
Are you normalizing your images to the 0-1 range?

**Comment by youmi-zym** [comment]
I have normalized my images to the 0-1 range when I load it.
`

            right_image = right_image - np.mean(right_image)

            right_image = right_image / np.std(right_image)`

and the code blow calculate the **L1_loss**
`

    self.l1_left=torch.abs(self.left_est-left)

    self.l1_reconstruction_loss_left=self.l1_left.mean()

    self.l1_right=torch.abs(self.right_est-right)

    self.l1_reconstruction_loss_right=self.l1_right.mean()`

**Comment by mrharicot** [comment]
You actually zero-centered and normalized your image, which is quite different.
This is a good practice before feeding the image to the neural network, however for the image reconstruction loss I usually only normalize my images to the 0-1 range by dividing by 255 if they are in 8bit.

**Comment by youmi-zym** [comment]
Thanks for your reply. The problem have solved. You are so great. I'm from China, a Frist-year graduate student, and recently I want to get a doctorate in foreign country. Is there any chance to join your group?

**Comment by mrharicot** [comment]
Hi,
The images are normalized by the function tf.convert_image_dtype:
https://github.com/mrharicot/monodepth/blob/master/monodepth_dataloader.py#L112
